sklearn/linear_model/tests/test_least_angle.py

```diff
@@ -135,27 +181,62 @@ def test_collinearity():
     assert_array_almost_equal(coef_path_, np.zeros_like(coef_path_))
 
 
+def test_collinearity_sufficient_stats():
+    # Check that lars_path is robust to collinearity in input
+    X = np.array([[3., 3., 1.],
+                  [2., 2., 0.],
+                  [1., 1., 0]])
+    y = np.array([1., 0., 0])
+    G = np.dot(X.T, X)
+    Xy = np.dot(X.T, y)
+    _, _, coef_path_ = ignore_warnings(linear_model.lars_path_gram)(
+        Xy=Xy, Gram=G, n_samples=y.size, alpha_min=0.01)
+    assert not np.isnan(coef_path_).any()
+    residual = np.dot(X, coef_path_[:, -1]) - y
+    assert_less((residual ** 2).sum(), 1.)  # just make sure it's bounded
+
+
 def test_no_path():
     # Test that the ``return_path=False`` option returns the correct output
+    alphas_, _, coef_path_ = linear_model.lars_path(
+        X, y, method='lar')
+    alpha_, _, coef = linear_model.lars_path(
+        X, y, method='lar', return_path=False)
 
-    alphas_, active_, coef_path_ = linear_model.lars_path(
-        diabetes.data, diabetes.target, method="lar")
-    alpha_, active, coef = linear_model.lars_path(
-        diabetes.data, diabetes.target, method="lar", return_path=False)
+    assert_array_almost_equal(coef, coef_path_[:, -1])
+    assert alpha_ == alphas_[-1]
 
+
+def test_no_path_sufficient_stats():
```

Comment:
I'm not sure that we get much value out of repeating these tests. Makes more sense to just ensure equivalence to lars_path for various data, using a loop or pytest parameterize

Reply:
Good point. I've simplified the tests.